components/header-bar/src/command-palette/hooks/use-navigation.js

```diff
@@ -77,13 +81,20 @@ export const useNavigation = ({ itemsArray, actionsArray }) => {
             const actionsListLength = actionsArray.length
 
             if (showGrid) {
+                const apps = itemsArray?.filter((item) => item.type === APP)
+                const expectedGridSize = gridColumns * gridRows
+                const appsGridSize =
+                    apps.length >= expectedGridSize
+                        ? expectedGridSize
+                        : apps.length
```

Comment:
This logic is repeated, perhaps it should be refactored or passed as an argument directly?